### ISSUE TITLE
Redefining a class, results in false errors afterwards

### ISSUE THREAD

**Issue by qantourisc** [PyCQA/pylint#993]
```
import json

class MyJsonEncoder(json.JSONEncoder):
	pass
A=MyJsonEncoder()
MyJsonEncoder=MyJsonEncoder()

def MyJsonHandlerA(value):
	return MyJsonEncoder.encode(value)

def MyJsonHandlerB(value):
	return A.encode(value)


class C(object):
	def stuff(self): return "A"

B=C()
A=C()

def AA():
	return A.stuff()
def BB():
	return B.stuff()

for i in [ AA(), BB(), MyJsonHandlerA({}) ]:
	print(i)
```

### Current behavior
return MyJsonEncoder.encode(value) => [no-value-for-parameter] No value for argument 'o' in unbound method call
return A.stuff() => [no-value-for-parameter] No value for argument 'self' in unbound method call

### Expected behavior
No errors (well not on these anyway)

### pylint --version output
Warning: option include-ids is deprecated and ignored.
Warning: option symbols is deprecated and ignored.
pylint 1.4.4, 
astroid 1.3.8, common 0.63.2
Python 2.7.9 (default, May 24 2015, 11:41:41) 
[GCC 4.8.3]

**Comment by PCManticore** [comment]
Hi,

Can you test with Pylint 1.6.1 and let me know if you still have this error?

**Comment by qantourisc** [comment]
1.6.1 works
